### PR TITLE
Update download URL

### PR DESCRIPTION
**Pull request by sl1pm4t**
The previous one seems to have stopped working (I'm getting 404).

### CODE REVIEW

File: bin/download [hunk 1]
```diff
@@ -34,7 +34,7 @@ download_golang() {
 
   platform=$(get_platform)
   arch=$(get_arch)
-  download_url="https://dl.google.com/go/go${version}.${platform}-${arch}.tar.gz"
+  download_url="https://storage.googleapis.com/golang/go${version}.${platform}-${arch}.tar.gz"
```
Comment (smorimoto) [inline bin/download:37]:
```suggestion
  download_url="https://go.dev/dl/go${version}.${platform}-${arch}.tar.gz"
```

Comment (sl1pm4t) [reply bin/download:37]: I tried that originally but that site didn't seem to have the sha256 available (in the expected file name format)

Comment (smorimoto) [reply bin/download:37]: Ah, I see.

**Comment by smorimoto** [comment]
The question here is, why can't CI find problem?

**Comment by smorimoto** [comment]
The code itself looks fine

**Review by smorimoto — APPROVED** [review]
The code itself looks fine

**Comment by kennyp** [comment]
I'm not having any issues with the current base URL either. :thinking:

**Comment by kennyp** [comment]
What architecture and version are you trying to install?

**Comment by sl1pm4t** [comment]
I was installing Darwin arm64.

**Comment by sl1pm4t** [comment]
It's possible the download site was only broken temporarily. I'll try the original again when I get back to my desk.

**Comment by kennyp** [comment]
Any update @sl1pm4t?

**Comment by sl1pm4t** [comment]
Seems to be working fine now.

**Comment by kennyp** [comment]
Awesome! I'm glad to hear it.

**Pull request closed without merging**